runtime/lua/vim/lsp.lua

```diff
@@ -1379,7 +1378,7 @@ function lsp.start_client(config)
       -- since 3.15.0
       clientInfo = {
         name = 'Neovim',
-        version = string.format('%s.%s.%s', version.major, version.minor, version.patch),
+        version = tostring(vim.version()),
```

Comment:
The spec doesn't say anything about requiring x.y.z exactly, so the full semver version is hopefully ok to report.